src/validators/literal.rs

```diff
@@ -20,269 +19,39 @@ impl BuildValidator for LiteralBuilder {
-            expected,
-            expected_repr,
-            name,
-        })
-    }
-}
-
-impl Validator for LiteralMultipleStringsValidator {
-    fn validate<'s, 'data>(
-        &'s self,
-        py: Python<'data>,
-        input: &'data impl Input<'data>,
-        extra: &Extra,
-        _slots: &'data [CombinedValidator],
-        _recursion_guard: &'s mut RecursionGuard,
-    ) -> ValResult<'data, PyObject> {
-        let either_str = input.validate_str(extra.strict.unwrap_or(false))?;
-        if self.expected.contains(either_str.as_cow()?.as_ref()) {
-            Ok(input.to_object(py))
-        } else {
-            Err(ValError::new(
-                ErrorType::LiteralError {
-                    expected: self.expected_repr.clone(),
-                },
-                input,
-            ))
-        }
-    }
-
-    fn get_name(&self) -> &str {
-        &self.name
-    }
-
-    fn complete(&mut self, _build_context: &BuildContext<CombinedValidator>) -> PyResult<()> {
-        Ok(())
-    }
-}
-
-#[derive(Debug, Clone)]
-pub struct LiteralMultipleIntsValidator {
-    expected: AHashSet<i64>,
-    expected_repr: String,
-    name: String,
-}
-
-impl LiteralMultipleIntsValidator {
-    fn new(expected_list: &PyList) -> Option<Self> {
-        let mut expected: AHashSet<i64> = AHashSet::with_capacity(expected_list.len());
-        let mut repr_args = Vec::new();
-        for item in expected_list.iter() {
-            if let Ok(int) = item.extract() {
-                expected.insert(int);
-                repr_args.push(int.to_string());
-            } else {
-                return None;
-            }
         }
-        let (expected_repr, name) = expected_repr_name(repr_args, "literal");
-        Some(Self {
-            expected,
-            expected_repr,
-            name,
-        })
-    }
-}
-
-impl Validator for LiteralMultipleIntsValidator {
-    fn validate<'s, 'data>(
-        &'s self,
-        py: Python<'data>,
-        input: &'data impl Input<'data>,
-        extra: &Extra,
-        _slots: &'data [CombinedValidator],
-        _recursion_guard: &'s mut RecursionGuard,
-    ) -> ValResult<'data, PyObject> {
-        let int = input.validate_int(extra.strict.unwrap_or(false))?;
-        if self.expected.contains(&int) {
-            Ok(input.to_object(py))
-        } else {
-            Err(ValError::new(
-                ErrorType::LiteralError {
-                    expected: self.expected_repr.clone(),
-                },
-                input,
-            ))
-        }
-    }
-
-    fn get_name(&self) -> &str {
-        &self.name
-    }
-
-    fn complete(&mut self, _build_context: &BuildContext<CombinedValidator>) -> PyResult<()> {
-        Ok(())
+        Ok(LiteralGeneralValidator::new(expected)?.into())
     }
 }
 
 #[derive(Debug, Clone)]
 pub struct LiteralGeneralValidator {
-    expected_int: AHashSet<i64>,
-    expected_str: AHashSet<String>,
+    expected_int: HashMap<i64, Py<PyAny>>,
```

Comment:
can we use `AHashMap` - it's generally significantly faster.

Reply:
That's what this is using, look at the import